llama_index/evaluation/guideline_eval.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional
+
+from langchain.output_parsers import PydanticOutputParser
+from pydantic import BaseModel, Field
+
+from llama_index.evaluation.base import BaseEvaluator, Evaluation
+from llama_index.indices.base import ServiceContext
+from llama_index.prompts.base import Prompt
+from llama_index.response.schema import Response
+
+
+class GuidelineEvaluator(BaseEvaluator):
```

Comment:
not sure if you've seen the work around constitutional ai (see langchain as an example), but what you're doing is somewhat similar. could be a good inspiration